README.md

```diff
@@ -71,115 +67,48 @@ The widget generator will include a **Gulpfile.js** and the necessary package.js
 
+#### BadgeWidgetBoilerplate
 
-### Gulp
+The badge boiler plate is a react fully developed mendix widget that Shows a value as a badge or a color label which is recomended for beginners. Its has the following features
 
-The following tasks are provided in the **Gulpfile.js** and can be started by running ``gulp <TASKNAME>`` or ``npm run <TASKNAME>``:
+* Display as a badge or a color label
+* Attach an onclick microflow and microflow
+* Set static data text when the dynamic data is not specified
 
-* ``default``
+#### Empty widget
 
-You can omit this taskname and just run ``gulp``. This will watch for any changes in your ``src`` directory. If a change occurs, it will automatically create a new **MPK** and copy this to your ``widgets`` directory.
+The empty template is a mendix react hello world widget recommended for more experienced developers.
 
-It will also copy any changed Javascript files to your deployment folder. This means you do not have to redeploy locally, you can just refresh your browser. For changes in ``XML`` or ``CSS`` you will still need to synchronize your project directory (press F4 in the modeler) and redeploy.
+Note: all boilerplate are designed based on typescript .tsx file extension.
+With .tsx files allow usae of all the JSX (JavaScript XML) code, JSX is an embeddable XML-like syntax. It is meant to be transformed into valid JavaScript. In order to use JSX you must do two things.
 
-* ``modeler``
+* Name your files with a .tsx extension
+* Enable the jsx option
 
-This will try to start the test-project that is included in the ``test`` folder. It automatically opens the Modeler with this project. If you have your test-project in a different location, you can change the following options in ``package.json``:
+### 2.4 Add unit tests for the widget ?
 
-```json
-  "paths": {
-    "testProjectFolder": "./test/",
-    "testProjectFileName": "Test.mpr"
-  },
-```
+If Yes is chosen unit tests are included in order to ensure individual units of the component are tested to determine whether they are fit for use. default value is `No`
 
-Note: If you provide a different path to the test-project in Windows, make sure you escape backslashes. So for example: ```C:\Projects\TestFolder``` needs to be properly written in the JSON as:
-
-```json
-    "testProjectFileName": "C:\\Projects\\TestFolder"
-```
+### 2.4 Add end to end tests for the widget ?
 
-* ``version``
+If Yes is chosen, end to end tests are included to ensure that the integrated components of an application function as expected. default value is `No`
 
-This task will show you the version number of the widget by reading the ``packages.xml`` file. You can set the version number with this task by typing:
+Note: Both `Unit` and `End to end` tests apply only to the BadgeWidgetBoilerplate.
 
-``gulp version -n=X.X.X`` or ``npm run version -- -n=X.X.X``
+The tool will then create Copied files, and run `npm install` to install development dependencies.
```

Comment:
Consider this re-write:  This tool will create a project folder with default files, and runs `npm install` to install development dependencies.